Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/ServiceFacadeInspector.cs

```diff
@@ -0,0 +1,405 @@
+                }
+            }
+            else if (typeof(BaseExtensionService).IsAssignableFrom(serviceType))
+            {
+                // Make sure the extension service profile isn't null
+                if (MixedRealityToolkit.Instance.ActiveProfile.RegisteredServiceProvidersProfile != null)
+                {
+                    // If this is an extension service, see if it uses a profile
+                    MixedRealityServiceConfiguration[] serviceConfigs = MixedRealityToolkit.Instance.ActiveProfile.RegisteredServiceProvidersProfile.Configurations;
+                    for (int serviceIndex = 0; serviceIndex < serviceConfigs.Length; serviceIndex++)
+                    {
+                        MixedRealityServiceConfiguration serviceConfig = serviceConfigs[serviceIndex];
+                        if (serviceConfig.ComponentType.Type.IsAssignableFrom(serviceType) && serviceConfig.ConfigurationProfile != null)
+                        {
+                            // We found the service that this type uses - draw the profile
+                            SerializedObject serviceConfigObject = new SerializedObject(MixedRealityToolkit.Instance.ActiveProfile.RegisteredServiceProvidersProfile);
+                            SerializedProperty serviceConfigArray = serviceConfigObject.FindProperty("configurations");
+                            SerializedProperty serviceConfigProp = serviceConfigArray.GetArrayElementAtIndex(serviceIndex);
+                            SerializedProperty serviceProfileProp = serviceConfigProp.FindPropertyRelative("configurationProfile");
+                            BaseMixedRealityProfileInspector.RenderProfile(serviceProfileProp, null, false, serviceType);
+                            EditorGUILayout.Space();
+                            foundAndDrewProfile = true;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            return foundAndDrewProfile;
+        }
+
+        /// <summary>
+        /// Gathers service types from all assemblies.
+        /// </summary>
+        private static void InitializeServiceInspectorLookup()
+        {
+            if (initializedServiceInspectorLookup)
+            {
+                return;
+            }
+
+            inspectorTypeLookup.Clear();
+
+            var typesWithMyAttribute =
+                 from assembly in AppDomain.CurrentDomain.GetAssemblies().AsParallel()
+                 from classType in assembly.GetTypes()
+                 let attribute = classType.GetCustomAttribute<MixedRealityServiceInspectorAttribute>(true)
+                 where attribute != null
+                 select new { ClassType = classType, Attribute = attribute };
+
+            foreach (var result in typesWithMyAttribute)
+            {
+                inspectorTypeLookup.Add(result.Attribute.ServiceType, result.ClassType);
+            }
+
+            initializedServiceInspectorLookup = true;
```

Comment:
>initializedServiceInspectorLookup = true; [](start = 12, length = 41)

Do you know if this implies that if I ever add new service inspectors, I need to reboot unity in order to pick them up (or close my project and reopen my project) or close my scene and reopen my scene?

It looks like this is doing a static one-time initialization, which is fine, and if this is gonna be the behavior we should just document this (i.e. in docs)

Reply:
It will be refreshed every time recompilation occurs.